Tools/autotest/arducopter.py

```diff
@@ -1448,6 +1449,67 @@ def fly_motor_fail(self, fail_servo=0, fail_mul=0.0, holdtime=30):
 
         self.do_RTL()
 
+    def fly_motor_vibration(self):
+        """Test flight with motor vibration"""
+        self.context_push()
+        self.set_parameter("INS_LOG_BAT_MASK", 3)
+        self.set_parameter("INS_LOG_BAT_OPT", 2)
+        self.set_parameter("LOG_BITMASK", 45054)
+        self.set_parameter("SIM_VIB_MOT_MAX", 350)
+        self.set_parameter("SIM_GYR_RND", 100)
+        self.set_parameter("SIM_DRIFT_SPEED", 0)
+        self.set_parameter("SIM_DRIFT_TIME", 0)
+        self.reboot_sitl()
+
+        self.takeoff(10, mode="LOITER")
+        self.change_alt(alt_min=50)
+
+        hover_time = 15
+        tstart = self.get_sim_time()
+        self.progress("Hovering for %u seconds" % hover_time)
+        while self.get_sim_time_cached() < tstart + hover_time:
+             attitude = self.mav.recv_match(type='ATTITUDE', blocking=True)
+        tend = self.get_sim_time()
+
+        self.do_RTL()
+        vibe_log = self.buildlogs_path("ArduCopter-vibration-log.bin")
+        self.log_download(vibe_log)
+        psd = self.mavfft_fttd(vibe_log, 1, 0, tstart * 1.0e6, tend * 1.0e6)
+        # ignore the first 20Hz
+        ignore_bins = 20
+        freq = psd["F"][numpy.argmax(psd["X"][ignore_bins:]) + ignore_bins]
+        if numpy.amax(psd["X"][ignore_bins:]) < -22 or freq < 200 or freq > 300:
+            raise NotAchievedException("Did not detect a motor peak, found %f at %f dB" % (freq, numpy.amax(psd["X"][ignore_bins:])))
+        else:
+            self.progress("Detected motor peak at %fHz" % freq)
+
+        # now add a notch and check that the peak is squashed
+        self.set_parameter("INS_NOTCH_ENABLE", 1)
+        self.set_parameter("INS_NOTCH_FREQ", freq)
+        self.set_parameter("INS_NOTCH_ATT", 50)
+        self.set_parameter("INS_NOTCH_BW", freq/2)
+        self.reboot_sitl()
```

Comment:
Need a try/catch block here as we don't automatically reboot SITL as part of the suite.  If the test dies we may not recover correctly.

Reply:
Ok